### DISCUSSION TITLE
Slightly different color between components

### DISCUSSION THREAD

**Discussion by aj3423** [nvim-lualine/lualine.nvim#680] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/4710875/168070931-44e3c327-a3c3-4286-8845-cc990d61e6b2.png)

My configuration:
```lua
use {
	'nvim-lualine/lualine.nvim',
	requires = { 'kyazdani42/nvim-web-devicons' },
	config = function()
		require('lualine').setup{
			sections = {
				lualine_y = {} -- remove scroll percentage
			}
		}
	end
}
```
It happens in both terminal and GUI. The system font is "Fira Code", I see some topics mention that it may caused by font, if so, is it possible to set font for LuaLine only?

**Comment by shadmansaleh** [comment]
Are you using termguicolors ?

try setting (ex->`:set termguicolors`, lua->`vim.opt.termguicolors = true` if not already

If that doesn't work Can you cd into your lualine installation open nvim and run the following commad ?
```vim
:lua =require'tests.statusline':snapshot()
```
What does the output say ?

And no lualine can't control what font is used its handled by terminal or gui you're using.

**Reply by aj3423** [reply]
@shadmansaleh Yes I did set `vim.opt.termguicolors = true` in *init.lua*, `echo &termguicolors` shows '1'

And `:lua =require'tests.statusline':snapshot()` shows:

![image](https://user-images.githubusercontent.com/4710875/168242058-9a51de97-009c-4510-be86-96d943a51723.png)

I tried search for the color "59c0ff" in some places but found nowhere:
- nvim plugin directory: *~/.local/share/nvim*
- nvim config directory: *~/.config/nvim*
- terminal config file: *kitty.conf*

My colorscheme is *doom-one*, I found that **any** other colorscheme works fine, and if I switch to any of them then switch back to *doom-one*, the problem is gone. And the `snapshot()` shows differently:
![image](https://user-images.githubusercontent.com/4710875/168257430-addee951-6f69-4fab-9587-1e0ad7b19876.png)

**Comment by shadmansaleh** [comment, marked as answer]
Are you setting your colorscheme before calling lualine.setup ? If not try that.

Can you link the doom-one colorscheme you're using ?

**Reply by aj3423** [reply]
It's caused by a conflict with plugin [colortunner](https://github.com/zefei/vim-colortuner/), it is used to adjust the contrast and brightness. I did adjust the contrast, so the main components looks a bit darker, but why not the separator, seems they are in different color?
I tried change the load order of colorscheme / lualine.setup / colortunner, same problem.
Here's the [doom-one](https://github.com/romgrk/doom-one.vim) I use, and a minimal *init.lua* to reproduce this:
```lua
require('packer').startup({function(use)
	use 'wbthomason/packer.nvim'

	use 'zefei/vim-colortuner'

	use {
		'nvim-lualine/lualine.nvim',
		requires = { 'kyazdani42/nvim-web-devicons' },
		config = function()
			require('lualine').setup{
				sections = {
					lualine_y = {} -- remove scroll percentage
				}
			}
		end
	}

	vim.opt.background = 'dark'
	vim.opt.termguicolors = true
	vim.g.doom_one_terminal_colors = true
	use 'romgrk/doom-one.vim'
	vim.cmd('color doom-one')
end})
```

**Reply by shadmansaleh** [reply]
> It's caused by a conflict with plugin [colortunner](https://github.com/zefei/vim-colortuner/), it is used to adjust the contrast and brightness. I did adjust the contrast, so the main components looks a bit darker, but why not the separator, seems they are in different color? 

That explains it. colortunner is changing highlight groups directly, but those highlight groups are suppose to be managed by lualine itself. When they are changed like this the internal highlight state and the actual colors that get displayed gets out of sync. The separator highlight groups are created on demand based on internal highlight state . I guess there's no solution to this lualine and colortunner are just not compatible with each other.
